source/common/http/http2/codec_impl.cc

```diff
@@ -888,16 +903,18 @@ void ConnectionImpl::sendSettings(const Http2Settings& http2_settings, bool disa
     ASSERT(rc == 0);
   }
 
+  const uint32_t initial_connection_window_size =
+      http2_options.initial_connection_window_size().value();
```

Comment:
This is not a parameter sent via the SETTINGS frame. Currently, the `Http2ProtocolOptions` specify both HTTP/2 "options" such as the initial window update as well actual SETTINGS parameters, and the custom parameters I am adding support for only apply to the latter.

I do think this could be made more explicit in the proto itself, but I would like to limit the scope of this PR and avoid a large API change.